include/aikido/common/pointers.hpp

```diff
@@ -0,0 +1,13 @@
+#ifndef AIKIDO_COMMON_POINTERS_HPP_
+#define AIKIDO_COMMON_POINTERS_HPP_
+
+#include <dart/common/SmartPointer.hpp>
+
+// clang-format off
+#define AIKIDO_DECLARE_POINTERS(X)                                             \
+  DART_COMMON_MAKE_SHARED_WEAK(X)                                              \
```

Comment:
Do we need to forward-declare `X` here, e.g. `class X`? Does `DART_COMMON_MAKE_SHARED_WEAK` do that internally?

I also suggest adding a comment that this macro _must be used in the class' namespace_. You'd end up with some pretty gross C++ code if you tried to use a fully qualified name, e.g. `Uniquedart::dynamics::BodyNodePtr`. 😉

Reply:
> Does DART_COMMON_MAKE_SHARED_WEAK do that internally?

Yes. https://github.com/dartsim/dart/blob/master/dart/common/SmartPointer.hpp#L42

> I also suggest adding a comment that this macro must be used in the class' namespace.

:+1: